pennylane/ops/qubit/parametric_ops.py

```diff
@@ -650,6 +650,524 @@ def control_wires(self):
 CPhase = ControlledPhaseShift
 
 
+class CPhaseShift00(Operation):
+    r"""
+    A qubit controlled phase shift.
+
+    .. math:: \mathtt{CPhaseShift00}(\phi) = \begin{bmatrix}
+                e^{i\phi} & 0 & 0 & 0 \\
+                0 & 1 & 0 & 0 \\
+                0 & 0 & 1 & 0 \\
+                0 & 0 & 0 & 1
+            \end{bmatrix}.
+
+    .. note:: The first wire provided corresponds to the **control qubit**.
+
+    **Details:**
+
+    * Number of wires: 2
+    * Number of parameters: 1
+    * Number of dimensions per parameter: (0,)
+    * Gradient recipe:
+
+    .. math::
+        \frac{d}{d \phi} \mathtt{CPhaseShift00}(\phi)
+        = \frac{1}{2} \left[ \mathtt{CPhaseShift00}(\phi + \pi / 2)
+            - \mathtt{CPhaseShift00}(\phi - \pi / 2) \right]
+
+    Args:
+        phi (float): rotation angle :math:`\phi`
+        wires (Sequence[int]): the wire the operation acts on
+        do_queue (bool): Indicates whether the operator should be
+            immediately pushed into the Operator queue (optional)
+        id (str or None): String representing the operation (optional)
+    """
+    num_wires = 2
+    num_params = 1
+    """int: Number of trainable parameters that the operator depends on."""
+
+    ndim_params = (0,)
+    """tuple[int]: Number of dimensions per trainable parameter that the operator depends on."""
+
+    grad_method = "A"
+    parameter_frequencies = [(1,)]
+
+    def generator(self):
```

Comment:
The generator is probably getting test coverage via `test_parameter_frequencies_match_generator`, which is trying to test parameter frequencies by comparing them against the generator.  We should probably also have a unittest for the `generator` method as well.

Reply:
Again, where would I put said test? Would it make sense creating a separate test class for generators? The tests existing tests are in `TestMultiRZ` and `TestPauliRot`which seem to contain all test for that op.

Reply:
Go ahead and create a new class.  Location doesn't matter overly much.